### DISCUSSION TITLE
All albums suddenly have the same images

### DISCUSSION THREAD

**Discussion by IPMAN890** [photoprism/photoprism#3245] [Q&A]
Not sure what happened as I haven't made any changes in photoprism for a while. I went into albums today and realised that all off them have the same set of images, seems like its been overwritten by another album or something. The thumbnails and image count are correct though, not sure what happened

Thanks

**Comment by graciousgrey** [comment]
Do you see any warnings or errors in the logs? Did you do manual changes to the database? What version are you running and how did you install PhotoPrism?

**Reply by IPMAN890** [reply]
no log errors i can see and haven't touched the database

current build: Build 221118-e58fee0fb

installed a while ago via docker in unraid

**Reply by graciousgrey** [reply]
We never saw this behaviour before and no one else reported something like this. Maybe re-indexing your library helps or shows some more log messages.

**Comment by electric** [comment]
I've also randomly experienced this, with nothing in particularly obvious in the logs and a re-index not doing anything.

The only "fix" that works seems to be deleting the album and re-creating it, manually re-adding each image - this is obviously much harder with large albums containing photos over a longer period of time.

**Reply by lastzero** [reply]
Could you test if upgrading to our development preview helps?

**Reply by graciousgrey** [reply]
@electric do you also use Unraid? @electric @IPMAN890 Do you use sqlite or mariadb?

**Reply by electric** [reply]
I use mariadb with TrueNAS Scale

Not entirely sure it's possible to use the preview version with my configuration, sorry

**Reply by lastzero** [reply]
Do you know which MariaDB version is installed? Sounds like a possible select or update query issue (assuming the browser cache has been cleared), but I've never seen anything like that.

**Reply by IPMAN890** [reply]
ive updated mine to dev preview but no change, mydb is using sqlite

**Reply by lastzero** [reply]
https://github.com/photoprism/photoprism/blob/develop/internal/query/covers.go

Let us know if you see any problem with this.

**Comment by barry-luijten** [comment]
I experience this same behaviour, plus, my albums suddenly all have the same name (Month Year)! When I open them, they all show the same set of photo's, none of which are part of any of the albums. 
When I rename an album, the title remains, but the contents keep being the same set of photo's.
I have noticed this after upgrading MariaDB from 10.10 to 10.11. I'm running fully dockerized on a Swarm cluster.
Looking at the files in storage/albums/album, they seem to contain the original album names and photo references. When I update the name of an album in Photoprism, I see the file is updated. When I create a new album, it shows up as another file in this directory and when opened, it shows the correct photos.
I'm not planning to re-create all of my albums, because that is a huge task.
I haven't digged into the database yet, but I'm guessing that the files in storage/albums/album are just cached versions of whatever is in the database.

**Reply by lastzero** [reply]
These are backups.

**Reply by lastzero** [reply]
Do you use any apps or scripts with PhotoPrism?

**Reply by barry-luijten** [reply]
No, not yet. I'm planning to use deck-chores to run regular cleanup/index jobs, which I'm already using to create database dumps in the mariadb container, but nothing in the photoprism container yet. Not using the api either.

**Reply by barry-luijten** [reply]
Can I use the backup files somehow to recreate the albums?

**Reply by lastzero** [reply]
Yes, they can unless they have been overwritten or deleted by now. A database backup can also help in this situation.

**Reply by lastzero** [reply]
Any chance you can tell us what you were doing before this "suddenly" happened? It's unlikely that it just happens in the background, it might rather be related to updating album metadata, adding or removing pictures?

**Comment by lastzero** [comment]
@barry-luijten @electric @IPMAN890 Note that we have created an issue for this so that we can gather details to reproduce the behavior and implement code changes if necessary:

- https://github.com/photoprism/photoprism/issues/3320

**Any help in providing a solution is greatly appreciated, as it allows our team to work on other issues that our community is waiting for. Thank you very much!**
